### PR TITLE
Containers: add slides and examples

### PR DESCRIPTION
**Pull request by leogermond**
- add vector example
- add map
- sets
- clarify containers
- example extract tool + container examples
- sorting
- gnatpp
- search

### CODE REVIEW

File: courses/fundamentals_of_ada/adv_250_containers.rst [hunk 1]
```diff
@@ -35,69 +35,255 @@ Introduction
...
+    - Different types of data containers
+    - Hold an :ada:`Element` type
+    - Container types are :ada:`tagged`
 
-* Sorting mechanisms
+* Types defined as a product of both
+
+    - A data structure
+    - An implementation
+    - Define some added operations
```
Comment (frank-at-adacore) [inline courses/fundamentals_of_ada/adv_250_containers.rst:49]: You said "both" and then listed 3
Not sure what you mean by line 49

File: courses/fundamentals_of_ada/adv_250_containers.rst [hunk 2]
```diff
@@ -35,69 +35,255 @@ Introduction
...
+=================
+
+-----------------------
+Data Structures (1/2)
+-----------------------
+
+* Vector
+
+   - Essentially an array
+   - :dfn:`Capacity` and size can differ
```
Comment (frank-at-adacore) [inline courses/fundamentals_of_ada/adv_250_containers.rst:66]: You should either explain the difference or, if explained later, remove it from here

File: courses/fundamentals_of_ada/adv_250_containers.rst [hunk 3]
```diff
@@ -35,69 +35,255 @@ Introduction
...
+.. include:: examples/containers/extracts/insert.adb
+    :code: Ada
 
-* Definite types
+-----------
+Iteration
+-----------
 
-   - More efficient
+* Container have a :ada:`Cursor` type
```
Comment (frank-at-adacore) [inline courses/fundamentals_of_ada/adv_250_containers.rst:237]: "has" not "have"

File: courses/fundamentals_of_ada/adv_250_containers.rst [hunk 4]
```diff
@@ -35,69 +35,255 @@ Introduction
...
+------
 
-* Every container has a type `Cursor` which points to an element in a container
+* Arrays
+
+   - `Ada.Containers.Generic_Array_Sort`
+   - `Ada.Containers.Generic_Constrained_Array_Sort`
 
-   - Points to the container as well as the element!
+* Any object that has indexing
```
Comment (frank-at-adacore) [inline courses/fundamentals_of_ada/adv_250_containers.rst:261]: Should we say "container object" or "object of a container type"?

File: courses/fundamentals_of_ada/adv_250_containers.rst [hunk 5]
```diff
@@ -35,69 +35,255 @@ Introduction
...
-* Hashed vs Ordered
+--------
+Search
+--------
 
-   - **Ordered** containers are sequential
+* Use :ada:`Find` for a :ada:`Cursor`
 
-      + Searching can be slow for large containers
+    - :ada:`<Pkg>.No_Element` is a :ada:`Cursor` if not found
```
Comment (frank-at-adacore) [inline courses/fundamentals_of_ada/adv_250_containers.rst:274]: Should read more like "No_Element" is an "empty" cursor for when Find is unsuccessful

**Review by frank-at-adacore — CHANGES_REQUESTED** [review]

File: courses/fundamentals_of_ada/adv_250_containers.rst (hunk 2 above)
Comment (leogermond) [reply courses/fundamentals_of_ada/adv_250_containers.rst:66]: Indeed, also size was improper, the most common name is length

**Review by frank-at-adacore — APPROVED** [review]

**Pull request merged**